pandas/tests/reshape/test_concat.py

```diff
@@ -1088,19 +1088,14 @@ def test_append_empty_frame_to_series_with_dateutil_tz(self):
         s = Series({"date": date, "a": 1.0, "b": 2.0})
         df = DataFrame(columns=["c", "d"])
         result = df.append(s, ignore_index=True)
-        # n.b. it's not clear to me that expected is correct here.
-        # It's possible that the `date` column should have
-        # datetime64[ns, tz] dtype for both result and expected.
-        # that would be more consistent with new columns having
-        # their own dtype (float for a and b, datetime64ns, tz for date).
         expected = DataFrame(
             [[np.nan, np.nan, 1.0, 2.0, date]],
             columns=["c", "d", "a", "b", "date"],
             dtype=object,
         )
-        # These columns get cast to object after append
         expected["a"] = expected["a"].astype(float)
         expected["b"] = expected["b"].astype(float)
+        expected["date"] = pd.to_datetime(expected["date"])
```

Comment:
can you add tests for empty EA types (period, interval, categorical), we might already have them but this seems like a good place to put them.

Reply:
as additional values in the Series? a scalar Period and scalar Interval. not sure what to do for categorical.

Reply:
doesn't work for interval or period. will raise an issue. This was fixed for datetime64[ns, tzutc()] 'by accident', when ensuring consistency between np.concatenate and concat_compat,  see https://github.com/pandas-dev/pandas/pull/35032#issuecomment-651032657

Reply:
it's a different issue for period and interval since concatenating list of Series doesn't work either, whereas for tz-aware datetime works on master. see OP

on master
```
>>> interval = pd.Interval(0, 1, closed="right")
>>> interval
Interval(0, 1, closed='right')
>>>
>>> period = pd.Period("2012", freq="A-DEC")
>>> period
Period('2012', 'A-DEC')
>>>
>>> s = pd.Series({"period": period, "interval": interval})
>>> s
period        2012
interval    (0, 1]
dtype: object
>>>
>>> df = pd.DataFrame(columns=["c"])
>>> df
Empty DataFrame
Columns: [c]
Index: []
>>>
>>> result = df.append([s, s], ignore_index=True)
>>> result
     c period interval
0  NaN   2012   (0, 1]
1  NaN   2012   (0, 1]
>>>
>>> result.dtypes
c           object
period      object
interval    object
dtype: object
>>>
>>> df = pd.DataFrame([s, s])
>>> df
  period interval
0   2012   (0, 1]
1   2012   (0, 1]
>>>
>>> df.dtypes
period        period[A-DEC]
interval    interval[int64]
dtype: object
>>>
```

Reply:
for period and interval, the issue is not limited to an empty dataframe

master
```
>>> df = pd.DataFrame([1, 2], columns=["c"])
>>> df
   c
0  1
1  2
>>>
>>> result = df.append([s], ignore_index=True)
>>> result
     c period interval
0  1.0    NaN      NaN
1  2.0    NaN      NaN
2  NaN   2012   (0, 1]
>>>
>>> result.dtypes
c           float64
period       object
interval     object
dtype: object
>>>
>>> result = df.append([s, s], ignore_index=True)
>>> result
     c period interval
0  1.0    NaN      NaN
1  2.0    NaN      NaN
2  NaN   2012   (0, 1]
3  NaN   2012   (0, 1]
>>>
>>> result.dtypes
c           float64
period       object
interval     object
dtype: object
>>>
```

> will raise an issue.

don't think necessary. I think covered by #22994 and #22957 and others linked in those issues.

Reply:
kk, yeah sometimes we have scattered tests and hard to tell whether we have exhaustive cases.